bvm/ballerina-observe/src/main/java/org/ballerinalang/observe/trace/SpanStore.java

```diff
@@ -50,11 +54,12 @@ void addSpan(String spanId, Map<String, Span> spanMap) {
         spans.put(spanId, spanMap);
     }
 
-    void addSpanContext(String spanId, Map<String, SpanContext> spanContextMap) {
-        spanContexts.put(spanId, spanContextMap);
-    }
-
     Map<String, Span> closeSpan(String spanId) {
+        spanContexts.remove(spanId);
         return spans.remove(spanId);
     }
+
+    public void addSpanContext(String spanId, Map<String, SpanContext> spanContextMap) {
```

Comment:
Wrong method name.

Reply:
Fixed in 4ffbfc6ad9a11cc24598f426352bc7c86b4b4724